bindings/python/iree/runtime/system_api.py

```diff
@@ -313,34 +321,48 @@ def instance(self) -> _binding.VmInstance:
     return self._instance
 
   @property
-  def modules(self) -> Modules:
-    return self._modules
+  def modules(self) -> BoundModules:
+    return self._bound_modules
 
   def create_function_abi(self, f: _binding.VmFunction) -> _binding.FunctionAbi:
     return self._vm_context.create_function_abi(self._config.device,
                                                 self._config.host_type_factory,
                                                 f)
 
-  def add_modules(self, modules):
+  def add_vm_modules(self, vm_modules):
     assert self._is_dynamic, "Cannot 'add_module' on a static context"
-    for m in modules:
-      name = m.name
-      if name in self._modules:
-        raise ValueError(f"Attempt to register duplicate module: '{name}'")
-      self._modules[m.name] = BoundModule(self, m)
-    self._vm_context.register_modules(modules)
+    for m in vm_modules:
+      if m.name in self._bound_modules:
+        raise ValueError(f"Attempt to register duplicate VmModule: '{m.name}'")
+      self._bound_modules[m.name] = BoundModule(self, m)
+    self._vm_context.register_modules(vm_modules)
 
-  def add_module(self, module):
-    self.add_modules((module,))
+  def add_vm_module(self, vm_module):
+    self.add_vm_modules((vm_module,))
 
 
-def load_modules(*modules, config: Optional[Config] = None):
-  """Loads modules into a new or shared context and returns them."""
-  context = SystemContext(modules=modules, config=config)
-  bound_modules = [context.modules[m.name] for m in modules]
+def load_vm_modules(*vm_modules, config: Optional[Config] = None):
+  """Loads VmModules into a new SystemContext and returns them."""
+  context = SystemContext(vm_modules=vm_modules, config=config)
+  bound_modules = [context.modules[m.name] for m in vm_modules]
   return bound_modules
 
 
-def load_module(module, config: Optional[Config] = None):
-  """Loads a module into a new or shared context and returns them."""
-  return load_modules(module, config=config)[0]
+def load_vm_module(vm_module, config: Optional[Config] = None):
+  """Loads a VmModule into a new SystemContext and returns it."""
+  return load_vm_modules(vm_module, config=config)[0]
+
+
+def load_vm_flatbuffer(vm_flatbuffer: bytes, backend: str) -> BoundModule:
+  """Loads a VM Flatbuffer into a callable module."""
+  vm_module = _binding.VmModule.from_flatbuffer(vm_flatbuffer)
+  config = Config(TARGET_BACKEND_TO_DRIVER[backend])
+  bound_module = load_vm_module(vm_module, config)
+  return bound_module
+
+
+def load_vm_flatbuffer_file(path: str, backend: str) -> BoundModule:
+  """Loads a file containing a VM Flatbuffer into a callable module."""
+  with open(path, "rb") as f:
```

Comment:
Can you leave a TODO here: `# TODO: There should be an API for mmap'ing the file which should be used instead of reading into memory.` (will make a big difference for large models)